pyop2/codegen/c/inverse.c

```diff
@@ -8,16 +8,34 @@ static PetscBLASInt ipiv_buffer[BUF_SIZE];
 static PetscScalar work_buffer[BUF_SIZE*BUF_SIZE];
 #endif
 
-static void inverse(PetscScalar* __restrict__ Aout, const PetscScalar* __restrict__ A, PetscBLASInt N)
+#ifndef PYOP2_INV_LOG_EVENTS
+#define PYOP2_INV_LOG_EVENTS
+PetscLogEvent ID_inv_memcpy = -1;
+PetscLogEvent ID_inv_getrf = -1;
+PetscLogEvent ID_inv_getri = -1;
+static PetscBool log_active_inv = 0;
```

Comment:
Does this need to be defined outside of the function body? I guess it prevents a tiny amount of memory allocation.

Reply:
Yes it doesn't need to be but it can so I thought I'd do that.